### PR TITLE
Just a code review :)

### PR DESCRIPTION
**Pull request by tomasdeml**
Hi,
I took the liberty of doing a small code review of your Eet library (just too much time on my hands :-)). I tried to make the code more .net-idiomatic and simplify it a bit. I also fixed a few issues with resource disposal. If you find the changes useful, feel free to integrate them.

Best regards,
TD

### CODE REVIEW

**Comment by jirihelmich** [comment]
Hi, thanks, this is really great. We will have a look and consider your suggestions. Some of the changes are, however, in conflict with our team practices, so we'll not integrate this fully.

Were there any bugs that you've discovered during your review and want to point out?

File: src/Mews.Eet/Communication/EetSoapClient.cs [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
+
+        public Task<SendRevenueXmlResponse> SendRevenueAsync(SendRevenueXmlMessage message)
+        {
+            return soapClient.SendAsync<SendRevenueXmlMessage, SendRevenueXmlResponse>(message, operation: "http://fs.mfcr.cz/eet/OdeslaniTrzby");
+        }
+
+        public void Dispose()
+        {
+            soapClient.Dispose();
+            x509certificate.Dispose();
```
Comment (jirihelmich) [inline src/Mews.Eet/Communication/EetSoapClient.cs]: > X509Certificate2 implements the IDisposable interface starting with the .NET Framework 4.6; in previous versions of the .NET Framework, the X509Certificate2 class does not implement this interface, and therefore the Dispose method does not exist.

Consider which platforms will be targeted before we incorporate this.

File: src/Mews.Eet/Communication/SoapHttpClient.cs [hunk 2]
```diff
@@ -1,31 +1,48 @@
...
+        }
+
+        public void Dispose()
+        {
+            httpClient.Dispose();
+        }
+
+        private static void EnableTls12()
+        {
+            ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls12;
```
Comment (jirihelmich) [inline src/Mews.Eet/Communication/SoapHttpClient.cs]: Thanks for this, it explains troubles with HTTP/HTTPS in some of the testing environments.

**Comment by tomasdeml** [comment]
Hit just one bug per se - HttpClient was failing when connecting to an endpoint secured with TLS 1.2 producol (https://github.com/MewsSystems/eet/pull/3/commits/1a670d2fe45f7f77bcbea2d02b1e92ba8e972bda).

**Comment by jirihelmich** [comment]
For start, I have incorporated your changes regarding async and await as I think that was the most problematic part of the library. Keeping the PR open as we need to have a look at least on Disposing.

**Comment by jirihelmich** [comment]
Creating an issue to review resource disposal. Thanks again for the review!

**Pull request closed without merging**